### DISCUSSION TITLE
Are code comment PRs welome?

### DISCUSSION THREAD

**Discussion by Abadulrehman** [SubtitleEdit/subtitleedit#7498] [Q&A]
Hey, I was just perusing the code to figure out how SubtitleEdit was doing one of its many nifty things, and I found it to be rather lacking in comments and function/method Summaries etc.

As someone with not much experience in C# or .NET, it was sometimes puzzling for me what was going on, and I've been writing comments/function summaries to help myself understand whats happening better.

And so I was wondering if PRs with comments/documentation-in-code welcome? Or does the code-base follow the code-_as_-documentation principle?

**Comment by niksedk** [comment]
Hi @Abadulrehman,

Code-as-documenation would be preferred, but many places in SE have way to many lines in methods. I've not really beautified the code much... going more for working stuff over polished code. I would prefer not having comment PRs, but a README.md here and there would be fine.
